src/hints/WotHHint.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import unknown from '../assets/hints/WotH_Unknown.png';
+import woth from '../assets/hints/WotH.png';
+import barren from '../assets/hints/barren.png';
+
+class WotHHint extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            current: 0,
+        };
+        this.images = [
+            unknown,
+            woth,
+            barren,
+        ];
+        this.altTexts = [
+            'Unknown',
+            'WotH',
+            'Barren',
+        ];
+        this.max = 2;
+        this.handleClick = this.handleClick.bind(this);
+    }
+
+    handleClick() {
+        this.setState((state) => ({ current: state.current < this.max ? state.current + 1 : 0 }));
```

Comment:
```suggestion
        this.setState((state) => ({ current: (state.current + 1) % this.images.length }));
```
less magic numbers, you can get rid of `this.max` then